saleor/dashboard/discount/forms.py

```diff
@@ -59,6 +59,9 @@ def __init__(self, *args, **kwargs):
 
     def clean(self):
         cleaned_data = super().clean()
+        if self.errors:
```

Comment:
Not a big deal, but I think a more common pattern in Django is to use `.get` to check if there are cleaned values:
```
discount_type = cleaned_data.get('type')
value = cleaned_data.get("value")

if type and value:
   # stuff 
```